### ISSUE TITLE
🐛 Bug: `no-local-dependencies`: should ignore dependencies listed in `bundle(d)Dependencies`

### ISSUE THREAD

**Issue by andreww2012** [JoshuaKGoldberg/eslint-plugin-package-json#2048]
### Bug Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have [searched for related issues](https://github.com/michaelfaith/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Prod dependencies using the `file:` protocol are correctly resolved by almost all package managers (except for the Yarn classic) during installation, if they are present in `bundle(d)Dependencies` array.

That's how [I've been declaring some dependencies in my package](https://npmx.dev/package-code/eslint-config-un/v/1.0.0-beta.12/package.json#L94) for some time (they were patched).

So such entries should not be flagged by this rule.

### Actual

Currently `foo` dependency is reported:

```json
{
  "dependencies": {
    "foo": "file:foo.tgz"
  },
  "bundleDependencies": ["foo"]
}
```

### Version

1.6.2

### ESLint Version

10.8.0

**Comment by michaelfaith** [comment]
Thanks for flagging this.  We should definitely support this use case 👍 Is this something you'd like to contribute a fix for?

**Comment by andreww2012** [comment]
I can work on the fix
